### ISSUE TITLE
Text colour doesn't adapt to live wallpaper colour.

### ISSUE THREAD

**Issue by RokeJulianLockhart** [8bitPit/Niagara-Issues#2806]
### Describe the bug

It used to be intermittent – possible to remediate by reinitialising Niagara – but it now never estimates it correctly.

### Steps to reproduce

1.  Install https://www.apkmirror.com/apk/peter-huber/niagara-launcher-%f0%9f%94%b9-fresh-clean/niagara-launcher-%f0%9f%94%b9-fresh-clean-1-12-4-release/niagara-launcher-%e2%80%a7-home-screen-1-12-4-android-apk-download/
1.  Set the font colour to adapt to the wallpaper colour:

    > ![Screenshot_20240717-215555](https://github.com/user-attachments/assets/dc66f235-af59-4fb4-953a-d998b0d339d7)

1.  Install https://f-droid.org/repo/com.github.cvzi.darkmodewallpaper_59.apk
1.  Apply #000000 as the wallpaper. (I have `#ffffff` set as the light mode wallpaper, and `#000000` as the dark mode wallpaper.)

### Does the issue also occur with other 3rd-party launchers?

Not applicable

### Other 3rd-party launcher(s) tested

I don't know of any alternative launchers which support this feature.

### Device name

Fairphone FP5

### Android version

Android 5.0 (Lollipop)

### Screenshots / Screen recordings

> ![Screenshot_20240717-215219](https://github.com/user-attachments/assets/6464335b-dbe8-402e-8d6d-4496f58e805a)

**Comment by TFackler** [comment]
Thanks for reporting this.

Just making sure, you have set the text color to adapt to the wallpaper, correct?
Because the screenshot you have provided shows the setting for the font to use rather than the color of it (Niagara Settings > Look > Text color).

**Comment by RokeJulianLockhart** [comment]
> https://github.com/8bitPit/Niagara-Issues/issues/2806#issuecomment-2260699740

@TFackler, yeah, I have it set to automatic. Sorry for uploading the wrong screenshot.

**Comment by TFackler** [comment]
I could reproduce the issue the other way around, so I got light text on the white wallpaper and light text on the dark wallpaper. While this was readable, it could have been better.

I will look into this eventually, but until then, there seems to be an easy workaround with the specific app you mentioned.

Namely, in the app, go to Advanced settings > scroll to the part with a grid about what theme and text color to use:
![image](https://github.com/user-attachments/assets/8d542e66-6478-4a84-b9c6-657e071031c4)

Set the text property to either dark or light depending on the wallpaper you will use for home/day and lock/day, as well as home/night and lock/night.

For me, I used white wallpaper for day and black wallpaper for night, so I would set home/day and lock/day to dark text (on white wallpaper) and set home/night and lock/night to light text (on black wallpaper).

I hope this helps. Fixing this issue will be a low priority. Past experience showed that fixing this is difficult because the API for getting wallpaper colors is very restrictive.

**Comment by RokeJulianLockhart** [comment]
> https://github.com/8bitPit/Niagara-Issues/issues/2806#issuecomment-2262406479

Indeed, those preferences work brilliantly. I'm really thankful. With such a simple workaround, indeed it's low priority.
